src/gap-skipper.js

```
@@ -0,0 +1,176 @@
+
+    this.timer_ = null;
+  }
+
+  /**
+   * Timer callback. If playback still has not proceeded, then we seek
+   * to the start of the next buffered region.
+   *
+   * @private
+   */
+  skipTheGap_(scheduledCurrentTime) {
+    let buffered = this.tech_.buffered();
+    let currentTime = this.tech_.currentTime();
+    let nextRange = Ranges.findNextRange(buffered, currentTime);
+
+    this.consecutiveUpdates = 0;
+    this.timer_ = null;
+
+    if (nextRange.length === 0 ||
+        currentTime !== scheduledCurrentTime) {
+      return;
+    }
+
+    this.logger_('skipTheGap_:',
+                 'currentTime:', currentTime,
+                 'scheduled currentTime:', scheduledCurrentTime,
+                 'nextRange start:', nextRange.start(0));
+
+    // only seek if we still have not played
+    this.tech_.setCurrentTime(nextRange.start(0) + Ranges.TIME_FUDGE_FACTOR);
+  }
+
+  /**
+   * Set a timer to skip the unbuffered region.
+   *
+   * @private
+   */
+  setTimer_() {
+    let buffered = this.tech_.buffered();
+    let currentTime = this.tech_.currentTime();
+    let nextRange = Ranges.findNextRange(buffered, currentTime);
+
+    if (nextRange.length === 0 ||
+        this.timer_ !== null) {
+      return;
+    }
+
+    let difference = nextRange.start(0) - currentTime;
+
+    this.logger_('setTimer_:',
+                 'stopped at:', currentTime,
+                 'setting timer for:', difference,
+                 'seeking to:', nextRange.start(0));
+
+    this.timer_ = setTimeout(this.skipTheGap_.bind(this),
+                             difference * 1000,
+                             currentTime);
+  }
+
+  /**
+   * A debugging logger noop that is set to console.log only if debugging
+   * is enabled globally
+   *
+   * @private
+   */
+  logger_() {}
+
+  /**
+   * A noop to ensure there is always have a dispose function even if there
+   * was no playerId in the global options and therefore the gapSkipper was
+   * never properly initialized
```

This sounds like `playerId` is a required parameter. You should throw an error instead of returning a broken object in this case.